src/api/core/warren/xapi.py

```diff
@@ -29,8 +29,7 @@ def add_date_column(statements: pd.DataFrame) -> pd.DataFrame:
         statements = statements.copy()
         # Disable chained assignment warning to make the transformation inplace
         with pd.option_context("mode.chained_assignment", None):
-            # Transform 'timestamp' column into a date with the format (YYYY-MM-DD)
-            statements["date"] = pd.to_datetime(statements["timestamp"]).dt.date
+            statements["date"] = pd.to_datetime(statements["timestamp"])
```

Comment:
Maybe we can directly parse and convert to timezone-aware UTC datetime objects?

```suggestion
            statements["date"] = pd.to_datetime(statements["timestamp"], utc=True)
```

Reply:
By default, Ralph output UTC timestamp. Adding `utc=True` would not solve the current issue. I need to parse the UTC timestamp back to the user timezone.